lib/logto_client.dart

```diff
@@ -132,8 +146,63 @@ class LogtoClient {
     }
 
     await _tokenStorage.save(
-        idToken: idToken,
-        accessToken: tokenResponse.accessToken,
-        refreshToken: tokenResponse.refreshToken);
+      idToken: idToken,
+      accessToken: tokenResponse.accessToken,
+      refreshToken: tokenResponse.refreshToken,
+    );
+  }
+
+  Future<void> signOut({
+    String? redirectUri,
+  }) async {
+    // Throw error is authentication status not found
+    final idToken = await _tokenStorage.idToken;
+
+    final httpClient = _httpClient ?? http.Client();
+
+    if (idToken == null) {
+      throw LogtoAuthException(
+          LogtoAuthExceptions.authenticationError, 'not authenticated');
+    }
+
+    try {
+      final oidcConfig = await _getOidcConfig(httpClient);
+
+      // Revoke refresh token if exist
+      final refreshToken = await _tokenStorage.refreshToken;
+
+      if (refreshToken != null) {
+        try {
+          await logto_core.revoke(
+            httpClient: httpClient,
+            revocationEndpoint: oidcConfig.authorizationEndpoint,
+            clientId: config.appId,
+            token: refreshToken,
+          );
+        } catch (e) {
+          // Do Nothing silently revoke the token
+        }
+      }
+
+      final postLogoutRedirectUri =
+          redirectUri == null ? null : Uri.parse(redirectUri);
+
+      final signOutUri = logto_core.generateSignOutUri(
+        endSessionEndpoint: oidcConfig.endSessionEndpoint,
+        idToken: idToken.serialization,
+        postLogoutRedirectUri: postLogoutRedirectUri,
+      );
+
+      await _tokenStorage.clear();
+
+      if (postLogoutRedirectUri != null) {
+        await FlutterWebAuth.authenticate(
+          url: signOutUri.toString(),
+          callbackUrlScheme: postLogoutRedirectUri.scheme,
+        );
+      }
+    } finally {
+      if (_httpClient == null) httpClient.close();
+    }
```

Comment:
Even with the Nullable postLogoutRedirectUri, we intend to run the Logto signout flow. We need to clean up the sign-in session under the logto domain as well.  Is it possible to call `FlutterWebAuth.authenticate` with a empty callbackUrlScheme?

Reply:
That could be why my sign in is not working. Because it is not cleaned up properly.

Reply:
Couldn't you just make a HTTP call to the url.

Reply:
I'm afraid that won't work. We need to clean up the logto session cookie for those webview pages. I guess it depends on whether the `FlutterWebAuth.authenticate` stores the cookie info on ios and android or not. 
However, you are right on native apps, truly we should not have to open a new webview with the signout endpoint to clean them. There should be a better way to clean the cookies of webviews directly without navigating. 

 Let me clean this up.